components/FilesViewerModal.tsx

```diff
@@ -0,0 +1,348 @@
+  border-radius: 999px;
+  height: 40px;
+  width: 40px;
+  margin: 16px;
+  cursor: pointer;
+
+  display: ${props => (props.disabled ? 'none' : 'block')};
+  ${props =>
+    props.direction &&
+    css`
+      position: absolute;
+
+      ${props.direction === 'left' ? 'left: 0' : 'right: 0'};
+    `}
+  &:hover {
+    background: ${themeGet('colors.primary.700')};
+  }
+  &:focus {
+    background: ${themeGet('colors.primary.700')};
+  }
+`;
+
+const Button = styled.button`
+  color: white;
+  border: none;
+  outline: none;
+  background: transparent;
+  z-index: 3000;
+  padding: 8px;
+  border-radius: 99px;
+  &:hover {
+    background: black;
+    color: white;
+  }
+  &:focus-visible {
+    background: black;
+    color: white;
+  }
+  cursor: pointer;
+`;
+
+const ButtonLink = styled.a`
+  display: block;
+  color: white;
+  border: none;
+  outline: none;
+  background: transparent;
+  z-index: 3000;
+  padding: 8px;
+  border-radius: 99px;
+
+  &:hover {
+    background: black;
+    color: white;
+  }
+
+  &:focus-visible {
+    background: black;
+    color: white;
+  }
+  cursor: pointer;
+`;
+
+const StyledImg = styled.img`
+  width: 100%;
+`;
+
+FilesViewerModal.propTypes = {
+  onClose: PropTypes.func.isRequired,
+  parentTitle: PropTypes.string,
+  /** Specifies what file url to open first (opens first if undefined) */
+  openFileUrl: PropTypes.string,
+  files: PropTypes.arrayOf(
+    PropTypes.shape({
+      url: PropTypes.string,
+      name: PropTypes.string,
+    }),
+  ),
+};
+
+const getFileName = url => {
+  const parts = url?.split('/');
+  return parts[parts?.length - 1];
+};
+
+const getFileTitle = file => {
+  if (!file) {
+    return null;
+  } else if (file.name) {
+    return file.name;
+  } else if (file.description) {
+    return file.description;
+  } else {
+    return getFileName(file?.url);
+  }
+};
+
+export default function FilesViewerModal({ onClose, parentTitle, files, openFileUrl }) {
+  const intl = useIntl();
+  const initialIndex = openFileUrl ? files?.findIndex(f => f.url === openFileUrl) : 0;
+  const [selectedIndex, setSelectedIndex] = useState(initialIndex);
+
+  const onArrowLeft = React.useCallback(() => setSelectedIndex(selectedIndex => Math.max(selectedIndex - 1, 0)), []);
+  const onArrowRight = React.useCallback(
+    () => setSelectedIndex(selectedIndex => Math.min(selectedIndex + 1, (files?.length || 1) - 1)),
+    [files],
+  );
+  useKeyBoardShortcut({ callback: onArrowRight, keyMatch: ARROW_RIGHT_KEY });
+  useKeyBoardShortcut({ callback: onArrowLeft, keyMatch: ARROW_LEFT_KEY });
+
+  const selectedItem = files?.length ? files[selectedIndex] : null;
+  const nbFiles = files?.length || 0;
+  const hasMultipleFiles = nbFiles > 1;
+  const contentWrapperRef = React.useRef(null);
+
+  const renderFile = ({ url, info, file }, contentWrapperRef) => {
+    let content = null;
+    const isText = endsWith(url, 'csv') || endsWith(url, 'txt');
+    const isPdf = endsWith(url, 'pdf');
```

Comment:
We can make that case-insensitive and a bit safer:

```suggestion
    const fileExtension = url?.split('.')?.pop()?.toLowerCase();
    const isText = ['csv', 'txt'].includes(fileExtension);
    const isPdf = fileExtension === 'pdf';
```